meta/include/gnuradio-4.0/meta/UncertainValue.hpp

```diff
@@ -0,0 +1,331 @@
+operator+=(T &lhs, const U &rhs) noexcept {
+    lhs = lhs + rhs;
+    return lhs;
+}
+
+template<typename T, typename U>
+    requires UncertainValueLike<T> || UncertainValueLike<U>
+[[nodiscard]] inline constexpr auto
+operator-(const T &lhs, const U &rhs) noexcept {
+    using ValueTypeT = UncertainValueType_t<T>;
+    using ValueTypeU = UncertainValueType_t<U>;
+    if constexpr (UncertainValueLike<T> && UncertainValueLike<U>) {
+        using ResultType = decltype(lhs.value - rhs.value);
+        if constexpr (complex_like<ValueTypeT> || complex_like<ValueTypeU>) {
+            // we are dealing with complex numbers -> use the standard uncorrelated calculation.
+            ResultType newUncertainty = { std::hypot(std::real(lhs.uncertainty), std::real(rhs.uncertainty)), std::hypot(std::imag(lhs.uncertainty), std::imag(rhs.uncertainty)) };
+            return UncertainValue<ResultType>{ lhs.value - rhs.value, newUncertainty };
+        } else {
+            // both ValueType[T,U] are arithmetic uncertainties
+            return UncertainValue<ResultType>{ lhs.value - rhs.value, std::hypot(lhs.uncertainty, rhs.uncertainty) };
+        }
+    } else if constexpr (UncertainValueLike<T> && arithmetic_or_complex_like<ValueTypeU>) {
+        return T{ lhs.value - rhs, lhs.uncertainty };
+    } else if constexpr (arithmetic_or_complex_like<ValueTypeT> && UncertainValueLike<U>) {
+        return U{ lhs - rhs.value, rhs.uncertainty };
+    } else {
+        static_assert(std::is_arithmetic_v<ValueTypeT> && std::is_arithmetic_v<ValueTypeU>);
+        return lhs - rhs; // unlikely to be called due to default '-' definition
+    }
+}
+
+template<UncertainValueLike T, typename U>
+inline constexpr T &
+operator-=(T &lhs, const U &rhs) noexcept {
+    lhs = lhs - rhs;
+    return lhs;
+}
+
+template<typename T, typename U>
+    requires UncertainValueLike<T> || UncertainValueLike<U>
+[[nodiscard]] inline constexpr auto
+operator*(const T &lhs, const U &rhs) noexcept {
+    using ValueTypeT = UncertainValueType_t<T>;
+    using ValueTypeU = UncertainValueType_t<U>;
+    if constexpr (UncertainValueLike<T> && UncertainValueLike<U>) {
+        using ResultType = decltype(lhs.value * rhs.value);
+        if constexpr (complex_like<ValueTypeT> || complex_like<ValueTypeU>) {
+            // we are dealing with complex numbers -> use standard uncorrelated calculation
+            ResultType newUncertainty = { std::hypot(std::real(lhs.value) * std::real(rhs.uncertainty), std::real(rhs.value) * std::real(lhs.uncertainty)),
+                                          std::hypot(std::imag(lhs.value) * std::imag(rhs.uncertainty), std::imag(rhs.value) * std::imag(lhs.uncertainty)) };
+            return UncertainValue<ResultType>{ lhs.value * rhs.value, newUncertainty };
+        } else {
+            // both ValueType[T,U] are arithmetic uncertainties
+            auto combinedUncertainty = std::hypot(lhs.value * rhs.uncertainty, rhs.value * lhs.uncertainty);
```

Comment:
Maybe this would be better (multiplication is great with relative errors, so going to relative and back to absolute):

```
    lhs.value * rhs.value * std::hypot(lhs.uncertainty / lhs.value, rhs.uncertainty / rhs.value);
```